src/prefect/agent/agent.py

```diff
@@ -187,7 +205,44 @@ def start(self) -> None:
                             )
                         )
         finally:
-            self.on_shutdown()
+            self.cleanup()
+
+    def setup(self) -> None:
+        self.agent_connect()
+
+        if self.api_address:
```

Comment:
Jw, is there a reason _not_ to have the API enabled by default?

Reply:
I think we want users to opt into this, since it's only useful for deployments. It also forces them to choose a port, which is good since any default port might be used by something else already.